app/javascript/application.js

```diff
@@ -93,11 +95,11 @@ class _LobstersFunction {
     this.curUser = null;
 
     this.storyFlagReasons = ({
-      <%= Vote::STORY_REASONS.map{|k,v| "'#{k}': '#{v}'" }.join(", ") %>
+      'O': 'Off-topic', 'A': 'Already Posted', 'B': 'Broken Link', 'S': 'Spam', '': 'Cancel'
```

Comment:
This (and `commentFlagReasons`) exist to avoid duplicating flags between the Vote model and here. Can you find another way to remove this duplication? Is there any negative to doing this with ERB?

Reply:
Ah, thanks for explaining.

It was already a project to remove inline scripts from the site (#696 #697 #779) as a long-term security improvement, so I don't want to add one back.

One option would be to have the js be the authoritative list of flag reasons and have Rails parse it out at startup. Maybe it's because I'm mostly a backend developer, but that feels backwards and I don't want to do that.

I guess we only add and remove flags very infrequently. Please add a comment to this js explaining the reasons are duplicated in the Vote model, and add the reverse comment to Vote model.

Reply:
Added the comments.

Reply:
I talked this through on stream and someone pointed out that I misunderstood your suggestion. An easy way to remove this duplication would be to have an invisible html element with the flags in json and have the js load them. Now that I understand that, I think that's the best idea. Sorry I had you write comments.

We don't need to make it a standalone helper please add a `meta` tag in the `head`.

So I think that'll be `<meta name="storyFlags" content="<%=  Vote::STORY_REASONS.to_json ">` and something very similar for `commentFlags`.

Reply:
I spent close to 2 hours making test work, but rspec is new to me. Will spend some more time later.